flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -230,30 +230,34 @@ function start_local_zk {
     done < <(grep "^server\." "${FLINK_DIR}/conf/zoo.cfg")
 }
 
-function wait_dispatcher_running {
+function wait_rest_endpoint_up {
+  local port=$1
+  local endpoint=$2
+  local name=$3
+  local regex=$4
   # wait at most 10 seconds until the dispatcher is up
-  local QUERY_URL="${REST_PROTOCOL}://${NODENAME}:8081/taskmanagers"
+  local QUERY_URL="${REST_PROTOCOL}://${NODENAME}:${port}/${endpoint}"
   local TIMEOUT=20
   for i in $(seq 1 ${TIMEOUT}); do
     # without the || true this would exit our script if the JobManager is not yet up
     QUERY_RESULT=$(curl ${CURL_SSL_ARGS} "$QUERY_URL" 2> /dev/null || true)
 
     # ensure the taskmanagers field is there at all and is not empty
-    if [[ ${QUERY_RESULT} =~ \{\"taskmanagers\":\[.+\]\} ]]; then
-      echo "Dispatcher REST endpoint is up."
+    if [[ ${QUERY_RESULT} =~ ${regex} ]]; then
+      echo "${name} REST endpoint is up."
       return
     fi
 
-    echo "Waiting for dispatcher REST endpoint to come up..."
+    echo "Waiting for ${name} REST endpoint to come up..."
     sleep 1
   done
-  echo "Dispatcher REST endpoint has not started within a timeout of ${TIMEOUT} sec"
+  echo "${name} REST endpoint has not started within a timeout of ${TIMEOUT} sec"
   exit 1
 }
 
 function start_cluster {
   "$FLINK_DIR"/bin/start-cluster.sh
-  wait_dispatcher_running
+  wait_rest_endpoint_up "8081" "taskmanagers" "Dispatcher" "\{\"taskmanagers\":\[.+\]\}"
```

Comment:
I think it would be reasonable to leave the function `wait_dispatcher_running` (but implement it using `wait_rest_endpoint_up`)